XMonad/Layout/DecorationEx/LayoutModifier.hs

```diff
@@ -0,0 +1,322 @@
+{-# LANGUAGE FlexibleInstances, MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Layout.DecorationEx.LayoutModifier
+-- Description :  Layout modifier which adds decorations to windows.
+-- Copyright   :  (c) 2007 Andrea Rossato, 2009 Jan Vornberger, 2023 Ilya Portnov
+-- License     :  BSD-style (see xmonad/LICENSE)
+--
+-- Maintainer  :  portnov84@rambler.ru
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- Layout modifier, which is responsible for creation of decoration rectangles
+-- (windows), updating and removing them when needed. It is parametrized by
+-- @DecorationGeometry@, which says where decorations should be placed, and by
+-- @DecorationEngine@, which says how decorations should look.
+-----------------------------------------------------------------------------
+
+module XMonad.Layout.DecorationEx.LayoutModifier (
+    -- * Usage
+    --
+    -- $usage
+    decorationEx,
+    DecorationEx
+  ) where
+
+import XMonad
+import XMonad.Prelude
+import qualified XMonad.StackSet as W
+import XMonad.Layout.LayoutModifier
+import XMonad.Layout.WindowArranger (diff, listFromList)
+import XMonad.Util.Invisible
+import XMonad.Util.XUtils hiding (paintTextAndIcons)
+
+import XMonad.Layout.DecorationEx.Common
+import XMonad.Layout.DecorationEx.Engine
+import XMonad.Layout.DecorationEx.Geometry
+
+-- $usage
+--
+-- This module exports @decorationEx@ function, which is a generic function for
+-- adding decorations to your layouts. It can be used to use different
+-- decoration geometries and engines in any combination.
+-- For most used combinations, there are convinience functions in
+-- "XMonad.Layout.DecorationEx.TextEngine", "XMonad.Layout.DecorationEx.TabbedGeometry",
+-- and "XMonad.Layout.DecorationEx.DwmGeometry".
+--
+-- You can use this module with the following in your
+-- @~\/.xmonad\/xmonad.hs@:
+--
+-- > import XMonad.Layout.DecorationEx.LayoutModifier
+-- Then edit your @layoutHook@ by adding the DwmStyle decoration to
```

Comment:
Hm, agreed. In fact if you look what does my "DwmGeometry" currently, it's more general than DWM. It is similar to DWM in that it draws decoration bar over the window, but it can place the bar on the left, on the right or in the center. But honestly I have no idea how to name it. Maybe you have ideas? :)